backend/routes/application/application.go

```diff
@@ -0,0 +1,97 @@
+package routes
+
+/*
+/api/v1/application
+GET: Retrieve a list of applications with the most recently updated statuses (or another subset, via query params)
+
+/api/v1/application/(id)
+GET: Retrieve details on a particular application
+
+ToDos (in future): Write Operations (POST, PUT, DELETE)
+*/
+
+import (
+	"errors"
+	"log"
+	"net/http"
+	"time"
+
+	restful "github.com/emicklei/go-restful/v3"
+)
+
+// Creating a REST layer for application - ApplicationResource
+
+type Application struct {
+	Id, Name string
+}
+
+type ApplicationResource struct {
+	applications map[string]Application
+}
+
+// Creating a webservice for application endpoints
+func (a ApplicationResource) Register(container *restful.Container) {
+	ws := new(restful.WebService)
+	ws.
+		Path("/api/v1/application").
+		Consumes(restful.MIME_JSON).
+		Produces(restful.MIME_JSON)
+
+	ws.Route(ws.GET("/{application-id}").To(a.findApplication))
+	ws.Route(ws.GET("").To(a.recentApplication))
+	container.Add(ws)
+}
+
+// GET Retrieve a list of applications with the most recently updated statuses
+func (a ApplicationResource) recentApplication(request *restful.Request, response *restful.Response) {
+	list := []Application{}
+	for _, each := range a.applications {
+		list = append(list, each)
+	}
+
+	err := response.WriteEntity(list)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// GET info of operations depening upon the id
+func (a ApplicationResource) findApplication(request *restful.Request, response *restful.Response) {
+	id := request.PathParameter("application-id")
+	app := a.applications[id]
+	if len(app.Id) == 0 {
+		response.AddHeader("Content-Type", "text/plain")
```

Comment:
We should also return error code 404 here, if the application isn't found. This is the standard HTTP error code that communicates that the resource is not available (with REST, the vast majority of errors are communicated via [HTTP codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes)).

Reply:
Yes, but if you check here at [L64](https://github.com/samyak-jn/managed-gitops/blob/d590aabc24810b13a0e03af2bb332b49ac7bc6ad/backend/routes/application/application.go#L64), there is an implementation of writing to an Error String which is returning `http.StatusNotFound` which is basically is

> const http.StatusNotFound untyped int = ### 404

Do you want something else, apart from this?

Reply:
Ah :sweat_smile:,  missed that, thanks!